lib/redis_cluster/pool.rb

```diff
@@ -54,8 +54,30 @@ def pipelined(args, &block)
       random_node.execute :pipelined, args, &block
     end
 
+    # Implements scan across all nodes in the pool.  Encodes the current node in the returned cursor.
+    # Cursors will behave strangely if the node list changes during iteration.
+    def scan(args)
+      cursor = args.first
+      options = args[1] || {}
+      orig_cursor, node_index = decode_scan_cursor(cursor)
+      next_cursor, result = @nodes[node_index].execute("scan", [orig_cursor.to_s, options])
+      [ encode_scan_cursor(next_cursor.to_i, node_index), result ]
+    end
+
     private
 
+    def encode_scan_cursor(scan_cursor, targeted_node)
+      if scan_cursor == 0
+        ((targeted_node + 1) % @nodes.size).to_s
```

Comment:
I don't understand this special case. What does 0 mean and why is the code so different? A comment above describing the cursor contract might be really helpful.

Reply:
I added a comment here to describe what is going on.     

"0" means the start and end of iteration.  So when we receive 0 from a node, we advance to the next node and start at the zero position.    The modulus causes this to loop back to the zero node when we advance past the last node.   The expression is also missing the `(position * nodes.size) `part because position is zero here so that drops away.